### PR TITLE
Distinguish between reporting probes and controlling probes

### LINKED ISSUE #1040: k8s: Pipes/controls don't work most of the time
See video: http://recordit.co/8BTwXcjnRO

It may (or may not) be k8s-specific since it works in my local k8s cluster.

Apart from the unexpected EOF error I also got a `Bad request` in pipes and a few `Probe _ is not connected right now...`

screen shot 2016-02-25 at 17 33 22
screen shot 2016-02-25 at 17 35 28

### PR DESCRIPTION
**Pull request by 2opremio**
Fixes #1040

### CODE REVIEW

**Comment by 2opremio** [comment]
It would be good to know why `go vet` and friends didn't catch this.

**Comment by 2opremio** [comment]
We should also distinguish between reporting hosts and containing hosts, otherwise we get strange stuff in the UI like:

screen shot 2016-02-26 at 02 06 28

(note the three hosts on the top). I will create a separate ticket for this.

**Comment by tomwilkie** [comment]
Its not go vet that catches that, its errcheck.

File: report/report.go [hunk 1]
```diff
@@ -209,4 +209,6 @@ const (
 	HostNodeID = "host_node_id"
 	// ProbeID is the random ID of the probe which generated the specific node.
 	ProbeID = "probe_id"
+	// ControlProbeID is the random ID of the probe which controls the specific node.
+	ControlProbeID = "control_probe_id"
```
Comment (tomwilkie) [inline report/report.go]: I don't think we should add another probe id (ControlProbeID ) to the metadata.  I think we should be more specific about what ProbeID is, and make sure its added correctly.

**Comment by tomwilkie** [comment]
I'd remove the extra ControlProbeID, its redundant - ProbeID is only used from controls, and it was getting added where it shouldn't - by the host tagger, on container nodes the kubernetes reported is reporting.

I would stop the host tagger from adding the ProbeID to all nodes that probe reports, and instead make the docker report add it to the nodes it controls.  And update the comment on ProbeID constant.  Does that make sense?

**Comment by tomwilkie** [comment]
I'm Tom and I approve of this change.

**Pull request merged**